Functions/Initialization/xASL_init_InitializeMutex.m

```diff
@@ -63,17 +63,99 @@
 
 %% --------------------------------------------------------
 %% 2) Initialize mutex object
+% xASL_GoNoGo initializes the x.mutex object and states, and it deals with
+% locking of the current module only
 x.mutex = xASL_GoNoGo(x.dir.LockDir);
+
+%% 3) Check if this module is locked by another process
 if ~x.mutex.Lock(x.settings.MUTEXID)
-    fprintf('Also, check that there is no filesystem permission issue\n');
 	fprintf(2,['ERROR in module_' x.ModuleName ': mutex is locked: %s in %s\n'], x.settings.MUTEXID, x.dir.LockDir);
-	fprintf('This means that this module is currently being parallel processed by another Matlab instance/worker\n');
+	fprintf('\n');
+    fprintf('This means that this module is currently being parallel processed by another Matlab instance/worker\n');
 	fprintf('If this is not the case, the locked folder needs to be removed before proceeding\n');
 	fprintf('Also, check that there is no filesystem permission issue\n');
     fprintf('Otherwise this error can be ignored\n');
-	return;
-end
+    fprintf('\n');
+
+    x.mutex.bAnyModuleLocked = true;
+else
+    
+    %% 4) Check if any module for this subject is locked
+    % Here we check if any other module for the same subject is locked.
+    % So whereas xASL_GoNoGo in step 2 above only checks the current
+    % module, here we check if ANY module for this subject is locked.
+
+    if ~strcmpi(x.ModuleName, 'population')
+
+        % If the current module is the population module, we don't need to
+        % perform this check because the population module is cannot be
+        % iterated over subjects, it is always run once.
+        % Plus, at the start of the population module itself, we already
+        % check if nWorkers>1, in which case we skip the population module
+        % (== the population module should not run in parallel).
+        if ~isfield(x, 'mutex') || ~isprop(x.mutex, 'Root')
+            warning('mutex field missing, something odd going wrong');
+        else
+            [~, subjectName] = fileparts(x.mutex.Root);
+            lockDir = fullfile(x.dir.xASLDerivatives, 'lock');
+						
+			% Go through all module that have lock dirs already and are thus potentially locked
+			checkModuleList = xASL_adm_GetFileList(lockDir, '.*', 'FPList', [], 1);
+            for iCheckModule=1:length(checkModuleList)
+                [~, checkModuleName] = fileparts(checkModuleList{iCheckModule});
 
+				% Verify that it is really a directory with an ASL module lock files
+                if strcmp(checkModuleName(1:12), 'xASL_module_')
+                    checkModuleName = checkModuleName(13:end); % Extract the name of the module
+                    if strcmpi(checkModuleName, x.ModuleName)

```

Comment:
Note that for the structural module, the name isn't xASL_module_Structural but xASL_module_T1.

This function will still work at the end, but it's not really too good to have this different name. I wasn't sure where to change that and I also wasn't sure if this wouldn't create more problems.

Reply:
It seems that the xASL_module_T1 name instead of xASL_module_Structural is given inside xASL_module_Structural.m and then it's only used in a few messages and inside this initi_InitializeMutex function - so we can safely change that to make the name consistent with the module name.

I did that already, so you can resolve this issue.